File: dotnet/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/Factory/Concrete/WindowsFactory.cs
using AbstractFactoryDesignPattern.Example1.Factory.Abstract;
using AbstractFactoryDesignPattern.Example1.Models.Abstract;
using AbstractFactoryDesignPattern.Example1.Models.Concrete;

namespace AbstractFactoryDesignPattern.Example1.Factory.Concrete;

public class WindowsFactory : GUIFactory
{
	public override Button CreateButton()
	{
		return new WindowsButton();
	}
	public override Menu CreateMenu()
	{
		return new WindowsMenu();
	}
	public override Dialog CreateDialog()
	{
		return new WindowsDialog();
	}
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/src/factory/concrete/WindowsFactory.ts
import { GUIFactory } from '../abstract/GUIFactory';
import { Button } from '../../models/abstract/Button';
import { Menu } from '../../models/abstract/Menu';
import { Dialog } from '../../models/abstract/Dialog';
import { WindowsButton } from '../../models/concrete/WindowsButton';
import { WindowsMenu } from '../../models/concrete/WindowsMenu';
import { WindowsDialog } from '../../models/concrete/WindowsDialog';

export class WindowsFactory extends GUIFactory {
  createButton(): Button { return new WindowsButton(); }
  createMenu(): Menu { return new WindowsMenu(); }
  createDialog(): Dialog { return new WindowsDialog(); }
}
